frontend/javascript/README.md

```diff
@@ -0,0 +1,713 @@
+// values are returned with a return statement
+function returnOne() {
+    let varToReturn = 1;
+    varToReturn += 1;
+    return varToReturn - 1;
+}
+// the value returned by a function is "slotted in" to the place where the
+// function was called
+// e.g. this is true, it looks like 1 === 1 to the interpreter since the
+// returnOne function returned 1
+returnOne() === 1;
+// if you omit a parameter when calling a function, its value is `undefined`
+// inside the function
+function identity(x) {
+    return x;
+}
+identity() === undefined;
+
+// if the interpreter has enough information to name a function itself, you can
+// omit the name from the initial declaration and later refer to it by the
+// context-assigned name
+let namedFunction = function() {
+    // I'm named namedFunction
+}
+namedFunction();
+let functionArray = [
+    function() { /* I'm named functionArray[0] */ },
+    function() { /* I'm named functionArray[1] */ },
+];
+functionArray[0]();
+functionArray[1]();
+
+// you can use arrow syntax functions (or 'lambdas') as more concise notation
+let otherNamedFunction = () => { /* I'm named otherNamedFunction */ };
+otherNamedFunction();
+// lambdas can do anything functions can, including taking parameters and
+// returning values
+let lambdaSuccessor = (number) => { return number + 1; };
+lambdaSuccessor(1) === 2;
+// if a lambda is able to return in one expression, you can omit the braces and
+// return statement as a shorthand. this function is equivalent to the last:
+lambdaSuccessor = (number) => number + 1;
+lambdaSuccessor(1) === 2;
+```
+
+### Objects
+Objects let you bundle a bunch of named values into one unit.
+```js
+// object literals are delimited by curly braces ({})
+let emptyObject = {};
+// you define properties on an object by providing a name and a value
+let greetingObject = {
+    // properties define values on an object
+    greeting: "Hello",
+    // nested objects are valid
+    nestedObject: {
+        nestedGreeting: "Hello from the inside",
+    },
+    // they can store functions to emulate methods
+    greet: function(name) {
+        // access the current object's properties with the `this` keyword
+        return `${this.greeting} ${user}.`;
+    },
+    // here's another way to define a method
+    greetExcited(name) {
+        return `${this.greeting} ${user}!`;
+    },
+}
+// properties are then accessed by writing the object's name, a dot (.), then
+// the property's name. this is called dot notation.
+greetingObject.greeting === "Hello";
+greetingObject.nestedObject.nestedGreeting === "Hello from the inside";
+greetingObject.greet("Alice") === "Hello Alice.";
+// you can modify an object outside of its declaration to change, remove, or add
+// properties
+greetingObject.greeting = "Goodbye";
+greetingObject.greetExcited = undefined;
+greetingObject.greetConfused = (name) => `${greetingObject.greeting} ${user}?`;
+greetingObject.greetConfused("Bob") === "Goodbye Bob?";
+```
+
+### Classes
```

Comment:
Should we include stuff on anonymous classes? Prototypes?

Reply:
Will leave for Nivedha or Avery's input.